.travis.yml

```diff
@@ -1,38 +1,28 @@
-language: python
-
-python:
-    - 2.7
+language: c
 
+os:
+    - osx
+    - linux
+    
 env: DISPLAY=:99 MOZ_XVFB=1
 
-before_install:
-    - sudo apt-get update
-    - "/sbin/start-stop-daemon --start --quiet --make-pidfile --pidfile /tmp/custom_xvfb_99.pid --background --exec /usr/bin/Xvfb -- :99 -ac -screen 0 1024x768x24"
-
 install:
-    - sudo apt-get install flashplugin-nonfree subversion
-
-    # As long as we have invasive changes lets get the trunk version of marionette-client
-    - svn checkout https://github.com/mozilla/gecko-dev/trunk/testing/marionette/client
-    - "cd client && python setup.py develop && cd .."
-
-    - python setup.py develop
-    - pip install mozdownload pep8
-
-before_script:
-    # Run pep8 on all except the checked out marionette-client folder
-    - pep8 --max-line-length=99 --exclude=client .
-
-    # Download Firefox Nightly which is compatible with the used version of
-    # marionette client
-    - mozdownload -t daily
-    - tar -xvf *firefox*.tar.bz2
+    - ./.travis/install.sh
 
 script:
-    - firefox-ui-tests --binary firefox/firefox
+    - ./.travis/run.sh
 
 notifications:
 #      email:
 #          - dev-automation@lists.mozilla.org
     irc:
-        - "irc.mozilla.org#automation"
\ No newline at end of file
+        - "irc.mozilla.org#automation"
+
+matrix:
+    allow_failures:
+      - os: osx
+    exclude:
+      - os: linux
+        compiler: clang
+      - os: osx
+        compiler: gcc
```

Comment:
We don't have compilers as requirement, and you do not define those above. So I think this excludes can be removed.